From b37614f0fa1a3407d8ce0416bde85424451f676a Mon Sep 17 00:00:00 2001
From: Daniel Krol <orblivion@gmail.com>
Date: Sat, 20 Nov 2021 13:15:46 -0500
Subject: [PATCH 11/20] Add sandstormPermissions to clientVars

---
 src/node/handler/PadMessageHandler.js | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/node/handler/PadMessageHandler.js b/src/node/handler/PadMessageHandler.js
index 5065a29c..807856eb 100644
--- a/src/node/handler/PadMessageHandler.js
+++ b/src/node/handler/PadMessageHandler.js
@@ -1003,6 +1003,10 @@ const handleClientReady = async (socket, message) => {
       return;
     }
 
+    const sandstormPermissions = process.env.SANDSTORM
+      ? decodeURIComponent(socket.client.request.headers["x-sandstorm-permissions"]).split(",")
+      : ["modify", "comment"];
+
     // Warning: never ever send sessionInfo.padId to the client. If the client is read only you
     // would open a security hole 1 swedish mile wide...
     const clientVars = {
@@ -1040,6 +1044,7 @@ const handleClientReady = async (socket, message) => {
       numConnectedUsers: roomSockets.length,
       readOnlyId: sessionInfo.readOnlyPadId,
       readonly: sessionInfo.readonly,
+      sandstormPermissions,
       serverTimestamp: Date.now(),
       userId: sessionInfo.author,
       abiwordAvailable: settings.abiwordAvailable(),
-- 
2.30.2

