From 414e80c8ed6bbdba3a7fd575a2c280c5521309c0 Mon Sep 17 00:00:00 2001
From: Daniel Krol <orblivion@gmail.com>
Date: Fri, 29 Oct 2021 15:28:53 -0400
Subject: [PATCH 11/25] Use Sandstorm permissions header for readonly

---
 src/node/hooks/express/webaccess.js | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/node/hooks/express/webaccess.js b/src/node/hooks/express/webaccess.js
index 8a183681..9dfb396c 100644
--- a/src/node/hooks/express/webaccess.js
+++ b/src/node/hooks/express/webaccess.js
@@ -42,6 +42,12 @@ exports.normalizeAuthzLevel = (level) => {
 };
 
 exports.userCanModify = (padId, req) => {
+  // SANDSTORM EDIT
+  if (process.env.SANDSTORM && decodeURIComponent(req.headers["x-sandstorm-permissions"])
+    .split(",")
+    .indexOf("modify") === -1)
+    return false
+  // END SANDSTORM EDIT
   if (readOnlyManager.isReadOnlyId(padId)) return false;
   if (!settings.requireAuthentication) return true;
   const {session: {user} = {}} = req;
-- 
2.20.1

