From 62b18f28d9ca1356647fcc0fc5953252e99e9f38 Mon Sep 17 00:00:00 2001
From: Daniel Krol <orblivion@gmail.com>
Date: Fri, 29 Oct 2021 14:34:20 -0400
Subject: [PATCH 12/26] Disable API

---
 src/node/handler/APIHandler.js | 29 +++++++++--------------------
 1 file changed, 9 insertions(+), 20 deletions(-)

diff --git a/src/node/handler/APIHandler.js b/src/node/handler/APIHandler.js
index 6bc6e537..89ca23be 100644
--- a/src/node/handler/APIHandler.js
+++ b/src/node/handler/APIHandler.js
@@ -30,20 +30,6 @@ const createHTTPError = require('http-errors');
 
 const apiHandlerLogger = log4js.getLogger('APIHandler');
 
-// ensure we have an apikey
-let apikey = null;
-const apikeyFilename = absolutePaths.makeAbsolute(argv.apikey || './APIKEY.txt');
-
-try {
-  apikey = fs.readFileSync(apikeyFilename, 'utf8');
-  apiHandlerLogger.info(`Api key file read from: "${apikeyFilename}"`);
-} catch (e) {
-  apiHandlerLogger.info(
-      `Api key file "${apikeyFilename}" not found.  Creating with random contents.`);
-  apikey = randomString(32);
-  fs.writeFileSync(apikeyFilename, apikey, 'utf8');
-}
-
 // a list of all functions
 const version = {};
 
@@ -158,12 +144,16 @@ exports.handle = async function (apiVersion, functionName, fields, req, res) {
     throw new createHTTPError.NotFound('no such function');
   }
 
-  // check the api key!
-  fields.apikey = fields.apikey || fields.api_key;
+  // SANDSTORM EDIT - Turning off API access completely
+  // NOTE: The earlier version of this API included what appeared to be an
+  // incomplete implementation of API permissions (including a typo in
+  // 'getRevisionsCount'). We assumed it was never used and didn't port it
+  // to this version. However if we want it for whatever reason, it can be
+  // found here:
+  //
+  // https://github.com/kentonv/etherpad-lite/commit/6150fe70e2856bbd3471e3a4d1a5109e97168dbd
 
-  if (fields.apikey !== apikey.trim()) {
-    throw new createHTTPError.Unauthorized('no or wrong API Key');
-  }
+  throw new createHTTPError.Unauthorized('Etherpad API not supported for Sandstorm');
 
   // sanitize any padIDs before continuing
   if (fields.padID) {
@@ -184,5 +174,4 @@ exports.handle = async function (apiVersion, functionName, fields, req, res) {
 };
 
 exports.exportedForTestingOnly = {
-  apiKey: apikey,
 };
-- 
2.20.1

