From d22593ea4d890bf38aada156ec5ef15b09097a21 Mon Sep 17 00:00:00 2001
From: Kenton Varda <kenton@sandstorm.io>
Date: Wed, 16 Nov 2016 16:12:12 -0800
Subject: [PATCH 07/12] Make authorship bars continuous through blank lines.

---
 static/js/index.js | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/static/js/index.js b/static/js/index.js
index 6d89c21..53be7a8 100644
--- a/static/js/index.js
+++ b/static/js/index.js
@@ -104,6 +104,9 @@ function authorViewUpdate($node, lineNumber, prevAuthor, authorClass){
       }
     }
   }
+  if (authorClass == 'none') {
+    authorClass = prevAuthor;
+  }
   if (prevAuthor === authorClass) {
     $authorContainer.addClass('concise');
   } else {
-- 
2.37.3

